slm_porosity/heat_transfer.f90

```diff
@@ -826,4 +867,54 @@ FUNCTION Spline_cubic (r_p, l_p, fr_p, fl_p, dfr_p, dfl_p)
       - fr_p*l_p**3 - r_p**2*(3*fl_p*l_p - dfl_p*l_p**2 + dfr_p*l_p**2))/(r_p-l_p)**3
   END FUNCTION Spline_cubic
 
+  FUNCTION three_parameter_model (temperature, liquid_fraction, Dsolid, Dliquid, fusion_jump)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) ::  temperature(:), liquid_fraction(:)
+    REAL (pr) :: three_parameter_model(SIZE(temperature))
+    REAL (pr) :: Dsolid, Dliquid, fusion_jump
+
+    three_parameter_model = 1.0_pr + Dsolid*temperature + &
+      (fusion_jump + (Dliquid - Dsolid)*(temperature - 1.0_pr))*liquid_fraction
+  END FUNCTION three_parameter_model
+
+  FUNCTION Dthree_parameter_model (temperature, liquid_fraction, Dtemperature, Dliquid_fraction, Dsolid, Dliquid, fusion_jump)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) :: Dtemperature(:), Dliquid_fraction(:), liquid_fraction(:), temperature(:)
+    REAL (pr) :: Dthree_parameter_model(SIZE(temperature))
+    REAL (pr) :: Dsolid, Dliquid, fusion_jump
+
+    Dthree_parameter_model = (fusion_jump + (Dliquid - Dsolid)*(temperature - 1.0_pr))*Dliquid_fraction + &
+      (Dliquid - Dsolid)*Dtemperature*liquid_fraction + Dsolid*Dtemperature
+  END FUNCTION Dthree_parameter_model
+
+  FUNCTION solidus_liquidus_enthalpy (temperature, a, b)
+    IMPLICIT NONE
+    REAL (pr) :: solidus_liquidus_enthalpy, temperature, a, b
+
+    solidus_liquidus_enthalpy = temperature*(a + b/2*temperature)
```

Comment:
First, according to the content, this function can't be connected with `solidus_liquidus`. Think over each term or name that you are going to introduce. Never use just a first idea. You must make a list of candidates in your mind and make a **conscious** choice. You main problem is the low level of consciousness. Watch for all your actions to train it.

Second, this function is single-purpose and, therefore, should be placed in the close proximity of `user_read_input` to facilitate reading of the code. Indeed, it should be nested function, but we deal with fortran not python.

Finally, I suggest to remove this function as almost useless and introduce `enthalpy(temperature, liquid_fraction)`. Then, your problem can be easily solved with the following pseudocode:
```
enthalpy_{S,L} = {0,1}
liquid_fraction_{S,L} = liquid_fraction(enthalpy_{S,L})
enthalpy_{S,L} = enthalpy(1 {-,+} delta_fusion/2, liquid_fraction_{S,L})
```